apps/sailfish/qml/components/DockedControls.qml

```diff
@@ -121,14 +125,52 @@ DockedPanel {
                 maximumValue: 100
 
                 onValueChanged: {
-                    kodi.volume = value
+                    if (passiveUpdate)
+                        return
+                    requestVolume(value)
+                }
+
+                // sometimes XBMC doesn't report a volumechanged event back after setting it
+                Timer {
+                    id: volumeChangeTimeout
+                    interval: 500
+                    repeat: false
+                    onTriggered: {
+                        // request volume again
+                        volumeSlider.volumeChangeOutstanding = false
+                        volumeSlider.requestVolume(volumeSlider.volumePending)
+                    }
                 }
 
-                Binding {
-                    target: volumeSlider
-                    property: "value"
-                    value: kodi.volume
+                Component.onCompleted: {
+                    kodi.onVolumeChanged.connect(processChangedVolume)
                 }
+
+                // send volume request, but don't flood xbmc
+                function requestVolume(value) {
+                    volumePending = value
+                    if (!volumeChangeOutstanding) {
+                        kodi.volume = value
+                        volumeChangeOutstanding = true
+                        volumeChangeTimeout.start()
+                    }
+                }
+
+                function processChangedVolume(volume) {
+                    // update UI
+                    passiveUpdate = true
+                    volumeSlider.value = volume
+                    passiveUpdate = false
+                    volumeChangeTimeout.stop()
+                    // send last queued volume change
+                    if (volumeChangeOutstanding) {
+                        volumeChangeOutstanding = false
+                        if (volumePending > -1 && volume !== volumePending) {
+                            requestVolume(volumePending)
+                        }
+                    }
+                }
+
```

Comment:
Doesn't the sailfish slider have a onReleased event? In Ubuntu I only send the volume there, so no flooding happens. I guess it's still nice to send some values if the user keeps the slider pressed.  but I guess that could be achieved much simpler just by doing:

Timer {
  interval: 500
  running: volumeSlider.pressed
  onTriggered: {
    kodi.volume = volumeSlider.value
  }

(I didn't check the Sailfish apis if the slier hast a pressed/released or similar, but that's probably how I'm gonna do it in Ubuntu)